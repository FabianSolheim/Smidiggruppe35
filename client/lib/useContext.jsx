import React from "react";
import { fetchJSON } from "./fetchJSON";
import { postJSON } from "./postJSON";

export const ApiContext = React.createContext({
    async postBooking(booking) {
        return await postJSON("/api/bookings", booking)
    }
});