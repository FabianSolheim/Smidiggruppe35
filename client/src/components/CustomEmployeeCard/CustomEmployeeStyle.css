
.buttonDiv {
    display: flex;
    justify-content: center;
    padding-bottom: 2rem;
    position: absolute;
    bottom: 0.3rem;
    left: 50%;
    transform: translateX(-50%);
}