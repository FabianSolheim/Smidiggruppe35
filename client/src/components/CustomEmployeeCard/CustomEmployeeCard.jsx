import { Avatar, Grid, Typography, Button } from "@mui/material";
import React from "react";
import "./CustomEmployeeStyle.css";
import Box from "@mui/material/Box";

export function CustomEmployeeCard({
  employee: { name, image, position, about, linkedin },
}) {
  return (
    <Box
      sx={{ borderRadius: 2, maxWidth: 350, height: 350, position: "relative"}}
      className="employeeCard"
      style={{ backgroundColor: "#DFE5E9" }}
    >
      <Box>
        <Grid
          container
          mt={2}
          gap={4}
          justifyContent="center"
          alignItems="center"
        >
          <Grid item container justifyContent="center" xs={2}>
            <Avatar sx={{ width: 56, height: 56 }} src={`${image}`} />
          </Grid>
          <Grid item container justifyContent="start" flexDirection={"column"} xs={8}>
              <Typography mb={0} variant="h6" color="primary">
                {name}
              </Typography>
              <Typography mb={0} paragraph color="textDisabled">
                {position}
              </Typography>
          </Grid>
        </Grid>
        <Box mt={3} pr={2} pl={2}>
          <Typography color="textPrimary" paragraph>
            {about}
          </Typography>
        </Box>
      </Box>
      <div className={"buttonDiv"}>
        <Button
          variant="outlined"
          target="_blank"
          href={linkedin}
          color="primary"
          style={{
            padding: "0 22px",
            textDecoration: "none",
            fontWeight: 500,
            borderRadius: 0,
            textTransform: "none",
          }}
          sx={{
            ":hover": {
              backgroundColor: "#034F7A",
              color: "white",
            },
          }}
        >
          LinkedIn
        </Button>
      </div>
    </Box>
  );
}
