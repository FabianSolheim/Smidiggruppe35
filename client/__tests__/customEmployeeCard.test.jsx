import React from "react";
import ReactDOM from "react-dom";
import { CustomEmployeeCard } from "../src/components/CustomEmployeeCard/CustomEmployeeCard";
import { MemoryRouter } from "react-router-dom";
import { useLoading } from "../lib/useLoader";

const testData = [
  {
    name: "test",
    image: "imageUrl",
    position: "test",
    about: "test",
    linkedin: "test",
  },
];

describe("customEmployeeCard test", () => {
  it("shows CustomEmployeeCard", () => {
    const domElement = document.createElement("Box");
    ReactDOM.render(
      <MemoryRouter>
        {testData.map((m) => (
          <CustomEmployeeCard key={m.name} employee={testData} />
        ))}
      </MemoryRouter>,
      domElement
    );
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
