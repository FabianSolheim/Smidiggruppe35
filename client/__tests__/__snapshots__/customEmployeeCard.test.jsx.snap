// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customEmployeeCard test shows CustomEmployeeCard 1`] = `"<div class=\\"employeeCard MuiBox-root css-v7jxda\\" style=\\"background-color: rgb(237, 230, 241);\\"><div class=\\"MuiBox-root css-0\\"><div class=\\"MuiGrid-root MuiGrid-container css-v0rbtg-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-2 css-y0g10o-MuiGrid-root\\"><div class=\\"MuiAvatar-root MuiAvatar-circular css-a2xf8u-MuiAvatar-root\\"><img src=\\"undefined\\" class=\\"MuiAvatar-img css-1pqm26d-MuiAvatar-img\\"></div></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-8 css-14mixqs-MuiGrid-root\\"><h4 class=\\"MuiTypography-root MuiTypography-h4 css-13d8583-MuiTypography-root\\"></h4><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-mzff8c-MuiTypography-root\\"></p></div><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><div class=\\"MuiBox-root css-1bbcutn\\"><h6 class=\\"MuiTypography-root MuiTypography-subtitle2 css-1vhmb89-MuiTypography-root\\"></h6></div></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item buttonDiv css-1gilgkt-MuiGrid-root\\"><button class=\\"MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-16mysmu-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" target=\\"_blank\\" style=\\"padding: 0px 22px; text-decoration: none; font-weight: 500; border-radius: 0; text-transform: none;\\">LinkedIn</button></div></div></div></div>"`;
